### PR TITLE
🐛 fix(sources): address OpenCode source adapter review issues

### PR DESCRIPTION
**Pull request by MagnumGoYB**
## Summary

Address code review issues for OpenCode source adapter and add MiMo model pricing.

## Requirement Classification

**Type:** Bugfix + Feature

**User-visible outcome:** Improved OpenCode adapter reliability + MiMo model cost estimation

**Target platform:** All platforms (darwin, linux, windows)

**Affected area:** internal/sources/opencode.go, internal/pricing/pricing.go

**Release decision:** Release required after merge

## Acceptance Criteria

1. ✅ Remove dead firstUserMessage field from opencodeSession struct
2. ✅ Add comprehensive test coverage for OpenCode edge cases (5 new unit tests)
3. ✅ Fix ?mode=ro connection string format for modernc.org/sqlite
4. ✅ Lower CJK title minimum threshold from <4 to <2 runes
5. ✅ Fix SQL query to handle malformed JSON gracefully
6. ✅ Add MiMo-V2.5 and MiMo-V2 series pricing (CNY)
7. ✅ Add MiMo pricing test coverage

## Changed Areas

- internal/sources/opencode.go: 46 lines changed
- internal/sources/sources_test.go: 288 lines added (5 new test functions)
- internal/pricing/pricing.go: 6 MiMo model entries added
- internal/pricing/pricing_test.go: 1 new test function

## Release Decision

**Release required** - Adds MiMo model pricing support.

## TDD / Test Evidence

**Test-first evidence:** All fixes were verified with failing tests before implementation.

**Acceptance criteria evidence map:**
1. Dead code removal: TestOpenCodeKeepsOriginalModelID verifies struct integrity
2. Edge case coverage: 5 new unit tests cover missing DB, malformed JSON, zero-token, non-assistant, thread name resolution
3. SQLite format: TestOpenCodeKeepsOriginalModelID verifies file: prefix works
4. CJK support: TestOpenCodeThreadNameResolution verifies short titles accepted
5. SQL robustness: TestOpenCodeSkipsMalformedJSON verifies graceful handling
6. MiMo pricing: TestDefaultCatalogCoversMiMoModels verifies all 6 model variants

**Failure/edge cases covered:**
- Missing database returns empty events (not error)
- Malformed JSON messages are skipped
- Zero-token messages are skipped
- Non-assistant messages are skipped
- Short CJK titles (< 4 runes) are accepted

## Validation

- [x] make check passes
- [x] make test passes
- [x] make build succeeds
- [x] make test-harness passes

## Skipped Validation

None - all validation steps completed.

## Risk and Rollback

**Residual risk:** Low - OpenCode changes are isolated; MiMo pricing is additive

**Rollback:** Revert commits c010bb0, 08020bd, d54f09d

## aitok Review Checklist

- [x] Requirement classification, acceptance criteria, and evidence map are filled.
- [x] Offline-first behavior is preserved; no hidden upload, sync, telemetry, or background network path was added.
- [x] No raw API key is read, printed, hashed, fingerprinted, or persisted.
- [x] Source adapters still stream local logs and include fixture coverage for malformed or missing fields.
- [x] CLI JSON/Markdown/table output remains stable or the change is documented and tested.
- [x] Release decision is explicit: feature/bugfix changes require follow-up release; repository-only engineering changes do not.
- [x] Bugfix PRs include a failing test, fixture, harness sensor, or clear manual reproduction.
- [x] Release/build impact is stated when dependencies, CI, or packaging changed.

### CODE REVIEW

File: internal/sources/opencode.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (o OpenCode) Read(ctx context.Context) ([]usage.UsageEvent, error) {
 
 func (o OpenCode) Scan(ctx context.Context, handle func(usage.UsageEvent) error) error {
 	dbPath := filepath.Join(o.Home, ".local", "share", "opencode", "opencode.db")
-	db, err := sql.Open("sqlite", dbPath+"?mode=ro")
+	db, err := sql.Open("sqlite", "file:"+dbPath+"?mode=ro")
```
[the remark this answers is not in the document]
Comment (MagnumGoYB) [reply internal/sources/opencode.go]: Fixed in commit 4a11728. Now using `url.PathEscape` to properly encode the database path in the `file:` URI, which handles paths with URI-reserved characters (`#`, `?`, etc.) correctly.
